clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/view/DcosJobProvider.groovy

```diff
@@ -20,16 +22,24 @@ class DcosJobProvider implements JobProvider<DcosJobStatus> {
 
   private final DcosClientProvider dcosClientProvider
   private final ObjectMapper objectMapper
+  private final AccountCredentialsProvider credentialsProvider
 
   @Autowired
-  DcosJobProvider(DcosClientProvider dcosClientProvider, ObjectMapper objectMapper) {
+  DcosJobProvider(DcosClientProvider dcosClientProvider, ObjectMapper objectMapper, AccountCredentialsProvider credentialsProvider) {
     this.dcosClientProvider = dcosClientProvider
     this.objectMapper = objectMapper
+    this.credentialsProvider = credentialsProvider
   }
 
   @Override
   DcosJobStatus collectJob(String account, String location, String id) {
-    def dcosClient = dcosClientProvider.getDcosClient(account, location)
+    def credentials = credentialsProvider.getCredentials(account)
+
+    if (!(credentials instanceof DcosAccountCredentials)) {
```

Comment:
AFAIK, this is one of the few places this happens where we would need to check for something like this. Operations already do this with the validators so those don't need it and it doesn't seem like any of the other providers need this.